### PR TITLE
API provides functionality for user to create order

### PR DESCRIPTION
**Pull request by dorianbuck**
PT: https://www.pivotaltracker.com/story/show/179716952
### User Story
```
As a restaurant API
In order for users to buy food they want from the restaurant
I would like to be able to create an order for a specific user
```
### Changes Proposed
- Cart class created to handle cart products in cart
- Cart product model created to handle orders from a authorized user
- Api response with a json message confirming product was added to cart. 
- Api response with a json message confirming if product was not added to cart.  
### What we learned working on this feature
- Better planing next time, before trying to set up a cart/order system that is not fully clear on beforehand
- good repetition on classes, db, migration, models, controllers, factory and reading a lot about db relations

### Screenshots (Videos)
image (1)
image (2)

### CODE REVIEW

**Review by tochman — APPROVED** [review]

File: app/controllers/api/carts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    cart_is_valid(cart)
+  end
+
+  private
+
+  def custom_error
+    render json: { message: 'Product not found!' }, status: 422
+  end
+
+  def cart_is_valid(cart)
```
Comment (tochman) [inline app/controllers/api/carts_controller.rb]: This method name should reflect what the method is actually doing. And it is rendering a response. So why not just call it `render_response`, or not extract it at all from the `create` method?

File: app/controllers/api/carts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    render json: {
+      message: 'This product was added to your cart!',
```
Comment (tochman) [inline app/controllers/api/carts_controller.rb]: This is not a blocker, but we did discuss that this message is a bit problematic when used on the client-side, didn't we? I would suggest it includes the product name (so we don't have to eff-around with `replace` on the client and give the user nice feedback. It's all about the UX. 

If you choose not to do this here, you should at least create a PT chore/feature for it.

**Review by tochman — CHANGES_REQUESTED** [review]

File: app/controllers/api/carts_controller.rb (hunk 1 above)
Comment (dorianbuck) [reply app/controllers/api/carts_controller.rb]: We have changed it to render_response

File: app/controllers/api/carts_controller.rb (hunk 2 above)
Comment (dorianbuck) [reply app/controllers/api/carts_controller.rb]: Thank you for clarifying this. We have implemented the right response

**Review by tochman — APPROVED** [review]
Good job

**Pull request merged**